tests/unittests/test_exception_handling.py

```diff
@@ -0,0 +1,117 @@
+    """Test cases to verify we get custom error messages when we do not recieve error from the API"""
+
+    client_obj = RechargeClient('test_access_token')
+    method = 'GET'
+    path = 'path'
+    url = 'url'
+
+    @parameterized.expand([
+        ['400_error', [400, RechargeBadRequestError], 'HTTP-error-code: 400, Error: The request was not understood by Recharge.'],
+        ['401_error', [401, RechargeUnauthorizedError], 'HTTP-error-code: 401, Error: The request was not able to be authenticated.'],
+        ['402_error', [402, RechargeRequestFailedError], 'HTTP-error-code: 402, Error: The request to the resource failed because of Payment issue.'],
+        ['403_error', [403, RechargeForbiddenError], 'HTTP-error-code: 403, Error: The request was authenticated but not authorized for the requested resource (Permission scope error).'],
+        ['404_error', [404, RechargeNotFoundError], 'HTTP-error-code: 404, Error: The requested resource was not found.'],
+        ['405_error', [405, RechargeMethodNotAllowedError], 'HTTP-error-code: 405, Error: The provided HTTP method is not supported by the URL.'],
+        ['406_error', [406, RechargeUnacceptableRequestError], 'HTTP-error-code: 406, Error: The request was unacceptable, or requesting a data source which is not allowed although permissions permit the request.'],
+        ['409_error', [409, RechargeConflictError], 'HTTP-error-code: 409, Error: The request is in conflict, or would create a conflict with an existing resource.'],
+        ['415_error', [415, RechargeJSONObjectError], 'HTTP-error-code: 415, Error: The request body was not a JSON object.'],
+        ['422_error', [422, RechargeUnprocessableEntityError], 'HTTP-error-code: 422, Error: The request was understood but cannot be processed due to invalid or missing supplemental information.'],
+        ['426_error', [426, RechargeInvalidAPI], 'HTTP-error-code: 426, Error: The request was made using an invalid API version.'],
+        ['429_error', [429, RechargeRateLimitError], 'HTTP-error-code: 429, Error: The request has been rate limited.'],
+        ['500_error', [500, RechargeInternalServiceError], 'HTTP-error-code: 500, Error: The request could not be processed due to internal server error.'],
+        ['501_error', [501, RechargeUnimplementedResourceError], 'HTTP-error-code: 501, Error: The resource requested has not been implemented in the current version.'],
+        ['502_error', [502, Server5xxError], 'HTTP-error-code: 502, Error: Unknown Error'],
+        ['503_error', [503, RechargeThirdPartyServiceTimeoutError], 'HTTP-error-code: 503, Error: A third party service on which the request depends has timed out.']
+    ])
+    @mock.patch('tap_recharge.client.RechargeClient.check_access_token')
+    @mock.patch('time.sleep')
+    @mock.patch('requests.Session.request')
+    def test_custom_exception_handling(self, name, test_data, expected_data, mocked_request, mocked_sleep, mocked_check_token):
```

Comment:
Here `name` parameter is unused in the test. Where is this getting used?

Reply:
The `name` contains the name for the test case. Hence, the test cases will be named like, `test_custom_exception_handling_0_400_error`, `test_custom_exception_handling_1_401_error` and so on.